### PR TITLE
feat: disable GraphiQL via feature flag

### PR DESCRIPTION
**Pull request by floric**
This sets the GraphiQL IDE features behind feature flags (`playground` and `graphiql`). Just disable the features if you don't use them to reduce the complexity and dependencies (`handlebars`). Maybe we could use make handlebars a  dependency only by moving the rendering of the template into a build script and providing a static template at runtime.

### CODE REVIEW

**Comment by maraisr** [comment]
maybe won't have to be breaking change, if graphiql is added to the default feature flags list?

**Comment by floric** [comment]
> maybe won't have to be breaking change, if graphiql is added to the default feature flags list?

Yes, would be fine for me as well as written here: https://github.com/async-graphql/async-graphql/pull/1182#issuecomment-1374914891 :)

If you prefer, I will add it as feature. My decision was based on the current README citing that all features are optional. I would adapt the README then accordingly.

File: integrations/actix-web/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ repository = "https://github.com/async-graphql/async-graphql"
 version = "5.0.5"
 
 [dependencies]
-async-graphql = { path = "../..", version = "5.0.5", default-features = false }
+async-graphql = { path = "../..", version = "5.0.5", default-features = false, features = ["graphiql"] }
```
[the remark this answers is not in the document]
Comment (floric) [reply integrations/actix-web/Cargo.toml:15]: This depends on the decision if we add graphiql as a default feature. But if we disable all features, we would still need to enable it anyway. Am I right?

File: src/http/mod.rs [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 use std::io::ErrorKind;
 
 use futures_util::io::{AsyncRead, AsyncReadExt};
+#[cfg(feature = "graphiql")]
 pub use graphiql_source::graphiql_source;
+#[cfg(feature = "graphiql")]
 pub use graphiql_v2_source::{Credentials, GraphiQLSource};
 use mime;
 pub use multipart::MultipartOptions;
+#[cfg(feature = "graphiql")]
```
[the remark this answers is not in the document]
Comment (floric) [reply src/http/mod.rs:21]: Oh, right. Tbh. I didn't think about the difference, but now I remember. Yes, I will use two different flags. But this leads to the question if both should be enabled by default? As they serve the same goal, it wouldn't make much sense.

[the remark this answers is not in the document]
Comment (sunli829) [reply src/http/mod.rs:21]: Yes, enabled by default 😄

[the remark this answers is not in the document]
Comment (floric) [reply src/http/mod.rs:21]: ok, I will separate it and enable both. :)

**Comment by floric** [comment]
> maybe won't have to be breaking change or rather less of a breaking change, if graphiql is added to the default feature flags list?

it's separated and enabled by default now.

File: integrations/actix-web/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (floric) [reply integrations/actix-web/Cargo.toml]: done, both are enabled by default now

**Comment by sunli829** [comment]
thanks! 🙂

**Pull request merged**